llvm/lib/CodeGen/BranchFolding.cpp

```diff
@@ -1422,8 +1422,7 @@ bool BranchFolder::OptimizeBlock(MachineBasicBlock *MBB) {
     // This has to check PrevBB->succ_size() because EH edges are ignored by
     // analyzeBranch.
     if (PriorCond.empty() && !PriorTBB && MBB->pred_size() == 1 &&
-        PrevBB.succ_size() == 1 && PrevBB.isSuccessor(MBB) &&
-        !MBB->hasAddressTaken() && !MBB->isEHPad()) {
```

Comment:
I remove `PrevBB.isSuccessor(MBB)` as it is redundant with `!PriorTBB` (no branch at the end of block). I hesitate about `MBB->isEHPad()`.